llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/VectorExt/IR/VectorExtOps.h

```diff
@@ -70,27 +69,70 @@ class DimensionalRange {
 // required during distribution.
 class LayoutIterator {
 public:
-  using State = llvm::MapVector<LayoutDimension, DimensionalIterator>;
-  using DimensionMapping =
-      llvm::DenseMap<int64_t, SmallVector<LayoutDimension>>;
-  void maybeFreezeAndConcatenate(const LayoutIterator &frozenIterator);
+  struct State {
+    SmallVector<int64_t> computeSIMTIndex() const;
```

Comment:
I think we have this implemented in GPUDistributionPatterns.cpp also

Reply:
Maybe we can remove that?

Reply:
yes looks to be a duplicate. I will remove that.